code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm

```diff
@@ -9,119 +9,68 @@
 
 	var/confirm = alert(usr, "Are you sure you want to completely reconstruct your form? This process can take up to twenty minutes, depending on how hungry you are, and you will be unable to move.", "Confirm Regeneration", "Yes", "No")
 	if(confirm == "Yes")
-		var/mob/living/carbon/human/C = src
-		var/nutrition_used = C.nutrition/2
+		chimera_regenerate()
 
-		if(C.reviving == 1) //If they're already unable to
-			C << "You are already reconstructing, or your body is currently recovering from the intense process of your previous reconstitution."
-			return
+/mob/living/carbon/human/proc/chimera_regenerate()
+	var/mob/living/carbon/human/C = src
```

Comment:
Why cast this? There's no need. If this is defined on human, you can just refer to everything with implicit things. Like one line below, just doing nutrition.